### ISSUE TITLE
Receiving voice quickly dies while sending continues to work

### ISSUE THREAD

**Issue by curlybryce** [henkelmax/simple-voice-chat#221]
**Please read the [FAQ](https://modrepo.de/minecraft/voicechat/faq) before submitting a bug report!**

**Bug description**
Transmitting my mic works all the time, for me and those I had tested with.  For me, receiving voice becomes intermittent then stops after a few seconds while it continues to work for everyone else.  I'm on Linux while the two I had tested with are on Windows.

**Steps to reproduce the issue**
1. Install Fabric in multimc, adding the fabric api and simple voice chat mods.
2. Connect to a server with the mods and a friend
3. Have the friend speak in-game for a few moments

**Expected behavior**
I am able to hear everyone instead of the audio quickly cutting out

**Log files**
[latest.log](https://gist.github.com/curlybryce/0a2246aaf1b788f7475ab63eb860dc88)

**Versions**
- Minecraft 1.17
- Fabric 1.17 - 0.11.3
- Mod 1.17 - 1.0.1

**Other mods**
N/A

**Screenshots (Optional)**
N/A

**Comment by henkelmax** [comment]
Can you send me the logs of the server?
Are only you experiencing these issues or does it happen for every other player too?

**Comment by curlybryce** [comment]
[server.log](https://gist.github.com/curlybryce/239ed49c33d20a365ddbccd8dd2679aa)
I'm the only one having issues.  My mic will still work (they can hear me) in game and everything works fine for the other two.  Their mics on my end are the only thing seeming to drop
Also, for reference, I'm chaotic_soup in the log

**Comment by henkelmax** [comment]
What are your system specs, operating system, internet speed, etc.?

**Comment by curlybryce** [comment]
**Specs**
CPU: AMD Ryzen 1700X
GPU: Radeon RX Vega 64
RAM: 16GB DDR4 3200
OS: Linux (Artix), Kernel: 5.11.16-artix1-1
Internet: 650Mbps down + 18Mbps up (never been any issues with dropped packets or anything from what I can tell)

**Java**
openjdk version "16.0.1" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-16.0.1+9 (build 16.0.1+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-16.0.1+9 (build 16.0.1+9, mixed mode, sharing)
Arguments in Minecraft: `-XX:+UseZGC -Xmx 8192M -Xms 512M`

Also, I've noticed the issue seems to occur whenever `[AudioChannelThread-$player-uuid/WARN]: Clearing buffers to avoid audio delay` appears in my client log

**Comment by henkelmax** [comment]
You stated that you were using version 1.0.1, but I think you are actually using version 1.0.3. There was a bug in my build system, that caused 1.0.2 and 1.0.3 still internally had 1.0.1 in the fabric.mod.json.
Can you try installing version 1.0.1 (The real version 1.0.1 :D) on your client?

The message `[AudioChannelThread-$player-uuid/WARN]: Clearing buffers to avoid audio delay` was introduced in 1.0.2.
It should fix the high audio delay some people had (it just cuts out some audio when your client is lacking behind), so I am curious if you are having a high voice chat delay if you are using version 1.0.1

**Comment by henkelmax** [comment]
Closed due to inactivity

**Comment by curlybryce** [comment]
Alrighty, sorry for lack of replies, I had gotten incredibly busy all of a sudden.  I looked a bit and it turns out I was using 1.0.4 (it had been reporting as 1.0.1).  Today I was able to test with a large amount of people (around 8).  First everything was upgraded to 1.2.3 which didn't seem to work at all for anyone, after which everyone downgraded to the actual 1.0.1 and everything worked perfectly (there was one I know of who was on 1.0.4 for their client and it worked perfect as well).
Just thought I'd give you an update.  The server and my client had been updated to use fabric api 0.36.0 as well
